### PR TITLE
feat: creator search

### PR DESCRIPTION
**Pull request by asiaziola**

### CODE REVIEW

File: src/gateway/router/routes/searchRoute.ts [hunk 1]
```diff
@@ -31,10 +31,20 @@ export async function searchRoute(ctx: Router.RouterContext) {
...
+          UNION ALL
+          SELECT 5 as sort_order, creator_id as id, 'creator' as type, '{}'::jsonb as interaction, '' as confirmation_status, '' as pst_ticker, '' as pst_name
+          FROM
+          (
+            WITH temp_creator AS (
+                SELECT DISTINCT owner as creator_id from interactions where owner ILIKE ?
+                UNION ALL
+                SELECT DISTINCT owner as creator_id FROM contracts WHERE owner ILIKE ?
+            )
+            SELECT DISTINCT * FROM temp_creator) AS creator
```
Comment (asiaziola) [inline src/gateway/router/routes/searchRoute.ts:43]: not satisfied with the query time, it takes a while to go through all the interactions :(

Comment (ppedziwiatr) [reply src/gateway/router/routes/searchRoute.ts:43]: try adding a gin index on the new 'owner' column
`CREATE INDEX idx_interaction_owner_gin ON interactions USING gin (owner gin_trgm_ops);`

**Review by ppedziwiatr — CHANGES_REQUESTED** [review]

File: src/gateway/router/routes/searchRoute.ts (hunk 1 above)
Comment (ppedziwiatr) [reply src/gateway/router/routes/searchRoute.ts:43]: and maybe also add it on the contracts table...

File: updates/2022_11_10_interactions_owner.sql [hunk 2]
```diff
@@ -0,0 +1,43 @@
+DROP INDEX interactions_source_index;
+DROP INDEX interactions_sort_key_index;
+DROP INDEX interactions_interaction_id_uindex;
+DROP INDEX interactions_evolve_index;
+DROP INDEX interactions_contract_id_index;
+DROP INDEX interactions_contract_id_block_height_conf_status_idx;
+DROP INDEX interactions_confirmation_status_index;
+DROP INDEX interactions_block_height_index;
+DROP INDEX idx_interaction_id_gin;
+DROP INDEX idx_interact_write_gin;
```
Comment (ppedziwiatr) [inline updates/2022_11_10_interactions_owner.sql:10]: `DROP INDEX interactions_last_sort_key_index; `

File: updates/2022_11_10_interactions_owner.sql [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    ON interactions (contract_id, block_height DESC NULLS LAST) INCLUDE (confirmation_status);
+
+CREATE INDEX interactions_confirmation_status_index
+    ON interactions (confirmation_status);
+
+CREATE INDEX interactions_block_height_index
+    ON interactions (block_height);
+
+CREATE INDEX idx_interaction_id_gin ON interactions USING gin (interaction_id gin_trgm_ops);
+CREATE INDEX idx_contract_id_gin ON interactions USING gin (contract_id gin_trgm_ops);
```
Comment (ppedziwiatr) [inline updates/2022_11_10_interactions_owner.sql:43]: `CREATE UNIQUE INDEX interactions_last_sort_key_index ON interactions (last_sort_key);`

**Review by ppedziwiatr — CHANGES_REQUESTED** [review]

**Pull request merged**